src/node/config.ts

```diff
@@ -345,108 +346,110 @@ export type ResolvedConfig = UserConfig & {
 
 const debug = require('debug')('vite:config')
 
-export async function resolveConfig(
-  mode: string,
-  configPath?: string
-): Promise<ResolvedConfig | undefined> {
-  const start = Date.now()
-  const cwd = process.cwd()
-  let config: ResolvedConfig | undefined
-  let resolvedPath: string | undefined
-  let isTS = false
-  if (configPath) {
-    resolvedPath = path.resolve(cwd, configPath)
-  } else {
-    const jsConfigPath = path.resolve(cwd, 'vite.config.js')
-    if (fs.existsSync(jsConfigPath)) {
-      resolvedPath = jsConfigPath
+export const resolveConfig = cache<Promise<ResolvedConfig | undefined>>(
```

Comment:
Guess we can omit this generic? I think typescript can auto infer it from the arg function